raster/src/main/scala/geotrellis/raster/io/geotiff/OverviewStrategy.scala

```diff
@@ -16,20 +16,97 @@
 
 package geotrellis.raster.io.geotiff
 
+import geotrellis.raster.CellSize
+
+import scala.collection.Searching._
+
+/**
+  * Options ported from GDAL (https://gdal.org/programs/gdalwarp.html#cmdoption-gdalwarp-ovr)
+  *  for automatic selection from among available overviews
+  */
 sealed trait OverviewStrategy
+
+object OverviewStrategy {
+  def DEFAULT = AutoHigherResolution
+
+  /**
+    * Select appropriate overview given the strategy.
+    *
+    * WARN: this function assumes that CellSizes are sorted. It interprets idx 0 as the position with the highest CellSize.
```

Comment:
Taking into the account that this is true, usage of the search function in the `selectIndexByProximity` function makes it really hard to interpret.